internal/service/workers/nooneisforgotten/main.go

```diff
@@ -55,6 +63,87 @@ func updatePassportScanEvents(db *pgdb.DB) error {
 	return nil
 }
 
+func claimPassportScanEvents(cfg config.Config) error {
+	evType := cfg.EventTypes().Get(evtypes.TypePassportScan, evtypes.FilterInactive)
+	if evType == nil {
+		return nil
+	}
+
+	db := cfg.DB().Clone()
+	events, err := pg.NewEvents(db).FilterByType(evtypes.TypePassportScan).FilterByStatus(data.EventFulfilled).Select()
+	if err != nil {
+		return fmt.Errorf("failed to select passport scan events: %w", err)
+	}
+
+	if len(events) == 0 {
+		return nil
+	}
+
+	eventsMap := make(map[string]data.Event, len(events))
+	nullifiers := make([]string, len(events))
+	for i, event := range events {
+		nullifiers[i] = event.Nullifier
+		eventsMap[event.Nullifier] = event
+	}
+
+	balances, err := pg.NewBalances(db).FilterByNullifier(nullifiers...).FilterDisabled().Select()
+	if err != nil {
+		return fmt.Errorf("failed to select balances for claim passport scan event: %w", err)
+	}
+
+	if len(balances) == 0 {
+		return nil
+	}
+
+	countryCodesMap := make(map[string]int64, len(balances))
+	for _, balance := range balances {
+		// normally should never happen
+		if balance.Country == nil {
+			return fmt.Errorf("balance have fulfilled passport scan event, but have no country")
+		}
+		if _, ok := countryCodesMap[*balance.Country]; !ok {
+			countryCodesMap[*balance.Country] = 0
+		}
+	}
+
+	countryCodesSlice := make([]string, 0, len(countryCodesMap))
+	for k := range countryCodesMap {
+		countryCodesSlice = append(countryCodesSlice, k)
+	}
+
+	countries, err := pg.NewCountries(db).FilterByCodes(countryCodesSlice...).Select()
+	if err != nil {
+		return fmt.Errorf("failed to select countries for claim passport scan events: %w", err)
+	}
+
+	for _, country := range countries {
+		if !country.ReserveAllowed || country.Reserved >= country.ReserveLimit {
+			delete(countryCodesMap, country.Code)
+			continue
+		}
+		countryCodesMap[country.Code] = country.ReserveLimit - country.Reserved
+	}
+
+	for _, balance := range balances {
```

Comment:
Can we drop confusing extra maps and slices? You could just find the country in `countries` slice and do all the checks in a single loop